Framework/Algorithms/test/Q1DWeightedTest.h

```diff
@@ -235,6 +238,77 @@ class Q1DWeightedTest : public CxxTest::TestSuite {
     TS_ASSERT_DELTA(result->y(0)[6], 251.052, 0.001);
   }
 
+  void testShapeTable() {
+    createShapeTable();
+
+    std::string outputWS = "q1d_shapes";
+    TS_ASSERT_THROWS_NOTHING(radial_average.setPropertyValue("InputWorkspace", m_inputWS))
+    TS_ASSERT_THROWS_NOTHING(radial_average.setPropertyValue("OutputWorkspace", outputWS))
+    TS_ASSERT_THROWS_NOTHING(radial_average.setPropertyValue("OutputBinning", "0.001,0.001,0.08"))
+    TS_ASSERT_THROWS_NOTHING(radial_average.setPropertyValue("WedgeWorkspace", outputWS + "_wedges"))
+    TS_ASSERT_THROWS_NOTHING(radial_average.setPropertyValue("AsymmetricWedges", "1"))
+    TS_ASSERT_THROWS_NOTHING(radial_average.setPropertyValue("ShapeTable", "MaskShapes"))
+
+    TS_ASSERT_THROWS_NOTHING(radial_average.execute())
+
+    WorkspaceGroup_sptr result;
+    TS_ASSERT_THROWS_NOTHING(result = std::dynamic_pointer_cast<WorkspaceGroup>(
+                                 AnalysisDataService::Instance().retrieve(outputWS + "_wedges")))
+    TS_ASSERT_EQUALS(result->getNumberOfEntries(), 2)
+  }
+
+  void testShapeTableResults() {
```

Comment:
This test is great, can you add one for asymmetric too? Don't forget to run the CompareWorkspaces at the end, and assert on the result.